src/app/pages/storage/disks/disk-form/disk-form.component.ts

```diff
@@ -107,29 +130,47 @@ export class DiskFormComponent {
     })
   }
 
+  preInit() {
+    this.aroute.params.subscribe(params => {
+      if (params['pk']) {
+        this.customFilter[0][0].push(params['pk']);
+      }
+    });
+  }
+
   afterInit(entityEdit: any) {
     this.ws.call('notifier.choices', ['HDDSTANDBY_CHOICES']).subscribe((res) => {
-      this.disk_hddstandby = _.find(this.fieldConfig, {name : 'disk_hddstandby'});
+      this.disk_hddstandby = _.find(this.fieldConfig, {name : 'hddstandby'});
       res.forEach((item) => {
         this.disk_hddstandby.options.push(
-            {label : item[1], value : item[0]});
+            {label : item[1], value : item[0].toUpperCase()});
       });
     });
 
     this.ws.call('notifier.choices', ['ADVPOWERMGMT_CHOICES']).subscribe((res) => {
-      this.disk_advpowermgmt = _.find(this.fieldConfig, {name : 'disk_advpowermgmt'});
+      this.disk_advpowermgmt = _.find(this.fieldConfig, {name : 'advpowermgmt'});
       res.forEach((item) => {
         this.disk_advpowermgmt.options.push(
-            {label : item[1], value : item[0]});
+            {label : item[1], value : item[0].toUpperCase()});
       });
     });
 
     this.ws.call('notifier.choices', ['ACOUSTICLVL_CHOICES']).subscribe((res) => {
-      this.disk_acousticlevel = _.find(this.fieldConfig, {name : 'disk_acousticlevel'});
+      this.disk_acousticlevel = _.find(this.fieldConfig, {name : 'acousticlevel'});
       res.forEach((item) => {
         this.disk_acousticlevel.options.push(
-            {label : item[1], value : item[0]});
+            {label : item[1], value : item[0].toUpperCase()});
       });
     });
   }
+
+  beforeSubmit(value) {
+    delete value.name;
+    delete value.serial;
+    delete value.passwd2;
+
+    value.critical = value.critical === '' ? null : value.critical;
```

Comment:
Nice!  I would suggest putting parens around the conditional part, just to make it more readable:
`value.critical = ( value.critical === '' ? null : value.critical );`
But either way, it's clearer than if statements.  Thanks!